packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/SyncClientResetIntegrationTests.kt

```diff
@@ -1180,23 +1182,30 @@ class SyncClientResetIntegrationTests {
 
     @Test
     fun recoverUnsyncedChanges_executeClientReset_pbs() = runBlocking {
-        performPbsTest { _, _, _, builder ->
-            recoverUnsyncedChanges_executeClientReset(builder)
+        performPbsTest { syncMode, app, user, builder ->
+            recoverUnsyncedChanges_executeClientReset(syncMode, app, user, builder)
         }
     }
 
     @Test
     fun recoverUnsyncedChanges_executeClientReset_flx() = runBlocking {
-        performFlxTest { _, _, _, builder ->
-            recoverUnsyncedChanges_executeClientReset(builder)
+        performFlxTest { syncMode, app, user, builder ->
+            recoverUnsyncedChanges_executeClientReset(syncMode, app, user, builder)
         }
     }
 
-    private fun recoverUnsyncedChanges_executeClientReset(builder: SyncConfiguration.Builder) {
+    private fun recoverUnsyncedChanges_executeClientReset(
+        syncMode: SyncMode,
+        app: TestApp,
+        user: User,
+        builder: SyncConfiguration.Builder
+    ) {
+        var testRealm: Realm? = null
         val channel = Channel<ClientResetEvents>(2)
         val config = builder.syncClientResetStrategy(object : RecoverUnsyncedChangesStrategy {
             override fun onBeforeReset(realm: TypedRealm) {
-                fail("Should not call onBeforeReset")
+                @Suppress("TooGenericExceptionThrown")
```

Comment:
No, this error is translated on the C++ side as well. This is now being triggered because we are using an actual client reset rather than the "fake" one.